DQMOffline/Trigger/plugins/PhotonMonitor.cc

```diff
@@ -0,0 +1,302 @@
+{
+  me.numerator   = ibooker.bookProfile(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, xmin, xmax, ymin, ymax);
+  me.denominator = ibooker.bookProfile(histname+"_denominator", histtitle+" (denominator)", nbinsX, xmin, xmax, ymin, ymax);
+}
+void PhotonMonitor::bookME(DQMStore::IBooker &ibooker, PhotonME& me, const std::string& histname, const std::string& histtitle, int nbinsX, double xmin, double xmax, int nbinsY, double ymin, double ymax)
+{
+  me.numerator   = ibooker.book2D(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, xmin, xmax, nbinsY, ymin, ymax);
+  me.denominator = ibooker.book2D(histname+"_denominator", histtitle+" (denominator)", nbinsX, xmin, xmax, nbinsY, ymin, ymax);
+}
+void PhotonMonitor::bookME(DQMStore::IBooker &ibooker, PhotonME& me, const std::string& histname, const std::string& histtitle, const std::vector<double>& binningX, const std::vector<double>& binningY)
+{
+  int nbinsX = binningX.size()-1;
+  std::vector<float> fbinningX(binningX.begin(),binningX.end());
+  float* arrX = &fbinningX[0];
+  int nbinsY = binningY.size()-1;
+  std::vector<float> fbinningY(binningY.begin(),binningY.end());
+  float* arrY = &fbinningY[0];
+
+  me.numerator   = ibooker.book2D(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, arrX, nbinsY, arrY);
+  me.denominator = ibooker.book2D(histname+"_denominator", histtitle+" (denominator)", nbinsX, arrX, nbinsY, arrY);
+}
+
+void PhotonMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+				 edm::Run const        & iRun,
+				 edm::EventSetup const & iSetup) 
+{  
+  
+  std::string histname, histtitle;
+
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+
+  histname = "photon_pt"; histtitle = "photon PT";
+  bookME(ibooker,photonME_,histname,histtitle,photon_binning_.nbins,photon_binning_.xmin, photon_binning_.xmax);
+  setTitle(photonME_,"Photon pT [GeV]","events / [GeV]");
+
+  histname = "photon_pt_variable"; histtitle = "photon PT";
+  bookME(ibooker,photonME_variableBinning_,histname,histtitle,photon_variable_binning_);
+  setTitle(photonME_variableBinning_,"Photon pT [GeV]","events / [GeV]");
+
+  histname = "photonVsLS"; histtitle = "photon pt vs LS";
+  bookME(ibooker,photonVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax,photon_binning_.xmin, photon_binning_.xmax);
+  setTitle(photonVsLS_,"LS","Photon pT [GeV]");
+
+  histname = "photon_phi"; histtitle = "Photon phi";
+  bookME(ibooker,photonPhiME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
+  setTitle(photonPhiME_,"Photon #phi","events / 0.1 rad");
+
+
+  histname = "photon_eta"; histtitle = "Photon eta";
+  bookME(ibooker,photonEtaME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
```

Comment:
Ok I am using the following one now, since we are mainly interested in barrel region for these, so looking at the ECAL tower size, 1.4442/0.087 =  ~17 bins, so I will cover -1.4442 to 1.4442 with 34 bins.